Tests expecting an AttributeError fail on Python 3.11

**Describe the bug**

On a pre-release of Python 3.11, tests expecting an `AttributeError` fail because the message associated with the `AttributeError` has changed.

**Expected behavior**

All tests pass.

**Steps To Reproduce**

Run the tests using a pre-release of Python 3.11. We saw this in Fedora Linux packaging while doing early Python 3.11 impact testing.

**Your environment**
Please run the following and paste the output.
```bash
$ python -c "import platform; print(platform.platform())"
Linux-5.16.12-200.fc35.x86_64-x86_64-with-glibc2.35.9000
$ python -c "import sys; print('Python ', sys.version)"
Python  3.11.0a6 (main, Mar  8 2022, 00:00:00) [GCC 12.0.1 20220222 (Red Hat 12.0.1-0)]
$ python -c "import pydicom; print('pydicom ', pydicom.__version__)"
pydicom  2.2.2
$ python -c "import pynetdicom; print('pynetdicom ', pynetdicom.__version__)"
(would be 2.0.1 if the package were successfully installed)
```

**Further details***

The failing tests are:

```
FAILED pynetdicom/tests/test_assoc_user.py::TestServiceUserAcceptor::test_mode_assignment_raises
FAILED pynetdicom/tests/test_assoc_user.py::TestServiceUserAcceptor::test_primitive_assignment_raises
FAILED pynetdicom/tests/test_assoc_user.py::TestServiceUserRequestor::test_mode_assignment_raises
FAILED pynetdicom/tests/test_assoc_user.py::TestServiceUserRequestor::test_primitive_assignment_raises
FAILED pynetdicom/tests/test_presentation.py::TestPresentationContext::test_as_scp
FAILED pynetdicom/tests/test_presentation.py::TestPresentationContext::test_as_scu
```

An example failure is:

```
_____________ TestServiceUserAcceptor.test_mode_assignment_raises ______________

self = <pynetdicom.tests.test_assoc_user.TestServiceUserAcceptor object at 0x7f4f023ba190>

    def test_mode_assignment_raises(self):
        """Test that assigning mode after init raises exception."""
        user = ServiceUser(self.assoc, mode="acceptor")
        assert user.mode == "acceptor"
        with pytest.raises(AttributeError, match=r"can't set attribute"):
>           user.mode = "requestor"
E           AttributeError: property 'mode' of 'ServiceUser' object has no setter

pynetdicom/tests/test_assoc_user.py:168: AttributeError

During handling of the above exception, another exception occurred:

self = <pynetdicom.tests.test_assoc_user.TestServiceUserAcceptor object at 0x7f4f023ba190>

    def test_mode_assignment_raises(self):
        """Test that assigning mode after init raises exception."""
        user = ServiceUser(self.assoc, mode="acceptor")
        assert user.mode == "acceptor"
>       with pytest.raises(AttributeError, match=r"can't set attribute"):
E       AssertionError: Regex pattern "can't set attribute" does not match "property 'mode' of 'ServiceUser' object has no setter".

pynetdicom/tests/test_assoc_user.py:167: AssertionError
```

Based on that output, I recommend changing each occurrence of

```python
with pytest.raises(AttributeError, match=r"can't set attribute"):
```

in `pynetdicom` to

```python
with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
```

I will offer a PR that implements this change.

--- a/pynetdicom/tests/test_assoc_user.py
+++ b/pynetdicom/tests/test_assoc_user.py
@@ -164,7 +164,7 @@ class TestServiceUserAcceptor:
         """Test that assigning mode after init raises exception."""
         user = ServiceUser(self.assoc, mode="acceptor")
         assert user.mode == "acceptor"
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.mode = "requestor"
 
         assert user.mode == "acceptor"
@@ -297,22 +297,22 @@ class TestServiceUserAcceptor:
         with pytest.raises(RuntimeError, match=msg):
             user.implementation_version_name = "1.2.3"
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.asynchronous_operations = (1, 1)
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.role_selection = {}
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.sop_class_common_extended = {}
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.sop_class_extended = {}
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.user_identity = "test"
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.extended_negotiation = []
 
     def test_add_neg_pre(self):
@@ -1477,7 +1477,7 @@ class TestServiceUserRequestor:
         """Test that assigning mode after init raises exception."""
         user = ServiceUser(self.assoc, mode="requestor")
         assert user.mode == "requestor"
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.mode = "acceptor"
 
         assert user.mode == "requestor"
@@ -1601,22 +1601,22 @@ class TestServiceUserRequestor:
         with pytest.raises(RuntimeError, match=msg):
             user.implementation_version_name = "1.2.3"
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.asynchronous_operations = (1, 1)
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.role_selection = {}
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.sop_class_common_extended = {}
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.sop_class_extended = {}
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.user_identity = "test"
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             user.extended_negotiation = []
 
     def test_accepted_common_raises(self):

--- a/pynetdicom/tests/test_presentation.py
+++ b/pynetdicom/tests/test_presentation.py
@@ -374,7 +374,7 @@ class TestPresentationContext:
         context = build_context("1.2.3")
         assert context.as_scp is None
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             context.as_scp = True
 
         context._as_scp = True
@@ -387,7 +387,7 @@ class TestPresentationContext:
         context = build_context("1.2.3")
         assert context.as_scu is None
 
-        with pytest.raises(AttributeError, match=r"can't set attribute"):
+        with pytest.raises(AttributeError, match=r"can't set attribute|has no setter"):
             context.as_scu = True
 
         context._as_scu = True

